akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1202,25 +1202,38 @@ trait FlowOps[+Out, +Mat] {
    *
    * '''Completes when''' upstream completes
    *
-   * '''Cancels when''' downstream cancels and substreams cancel
+   * '''Cancels when''' downstream cancels and substreams cancel if propagateSubstreamCancel=false, downstream
+   * cancels or any substream cancels if propagateSubstreamCancel=true
    *
    * See also [[FlowOps.splitAfter]].
    */
-  def splitWhen(p: Out ⇒ Boolean): SubFlow[Out, Mat, Repr, Closed] = {
+  def splitWhen(propagateSubstreamCancel: Boolean)(p: Out ⇒ Boolean): SubFlow[Out, Mat, Repr, Closed] = {
```

Comment:
Technically we started phasing out booleans with special classes to make the APIs more readable, so in javadsl it would follow that pattern instead of a boolean I think.